code/obj/machinery/espresso.dm

```diff
@@ -180,24 +197,44 @@
 			src.add_fingerprint(user)
 			if (src.my_carafe) //freaking spacing errors made me waste hours on this
 				if (!(status & (NOPOWER|BROKEN)))
-					switch (alert("What would you like to do with [src]?",,"Brew coffee","Remove carafe","Nothing"))
-						if ("Brew coffee")
-							for(var/obj/item/reagent_containers/food/drinks/carafe/C in src.contents)
-								C.reagents.add_reagent("coffee_fresh",100)
-								playsound(src.loc, 'sound/misc/pourdrink.ogg', 50, 1)
-						if ("Remove carafe")
-							if (!src.my_carafe)
-								user.show_text("The carafe is gone!")
+					if(!src.emagged)
```

Comment:
delete, because coffee and tea brewing doesn't need to be separated this much, using the changes below
```suggestion
```